### ISSUE TITLE
What exactly is a client?

### ISSUE THREAD

**Issue by taugwa** [nus-cs2030/2223-s2#123]
Hi! I know that the word "client" has been used a lot in lectures and sometimes in labs and I have an idea of what it means. But can I ask what exactly is it referring to? Is it referring to the Main class? Or is it referring to the person who will be using the code to carry out its function?

Thank you :)

**Comment by displeasedcat** [comment]
I believe "client" is referring to the person that will be using the code that you have created

**Comment by edwinhrj** [comment]
hello, I believe that when the professor mentions anything to do with a "client", he is describing a class that requests services from another class.

This client is not interested in how the other class goes about providing the requested service, but just uses the others class' required methods.

hope this answers your question! and feel free to let me know if there's any issues

**Comment by andersooi** [comment]
I think the Main class can be referred to as a client since, like what has been mentioned, it is executing a service (method) from another class, but it does not need to know about how the method itself works. It just needs to run it and get the desired output.

**Comment by taugwa** [comment]
alright thanks guys 🙏🏻🙏🏻

**Comment by geraldckj** [comment]
In layman terms, client is whoever is using a particular class or method. In the context of TDA, you want to hide the variables of a method/class you're calling from the thing calling it

**Comment by SimonSeeYongsheng** [comment]
The client is basically the class that is invoking the method of another class

**Comment by zhanyang01** [comment]
client is basically the higher class that will use the method of the other classes

**Comment by timothysashimi** [comment]
The client can be referred to classes that calls other classes' methods

**Comment by aldensiol** [comment]
Thank you for posting this! I was also clueless for awhile as to what "clients" were referring to.
